moveit_kinematics/srv_kinematics_plugin/src/srv_kinematics_plugin.cpp

```diff
@@ -233,98 +240,100 @@ bool SrvKinematicsPlugin::searchPositionIK(const std::vector<geometry_msgs::Pose
+                                                                       << ") in searchPositionIK");
     error_code.val = error_code.NO_IK_SOLUTION;
     return false;
   }
 
   // Create the service message
-  moveit_msgs::srv::GetPositionIK ik_srv;
-  ik_srv.request.ik_request.avoid_collisions = true;
-  ik_srv.request.ik_request.group_name = getGroupName();
+  auto ik_srv = std::make_shared<moveit_msgs::srv::GetPositionIK::Request>();
+  ik_srv->ik_request.avoid_collisions = true;
+  ik_srv->ik_request.group_name = getGroupName();
 
   // Copy seed state into virtual robot state and convert into moveit_msg
   robot_state_->setJointGroupPositions(joint_model_group_, ik_seed_state);
-  moveit::core::robotStateToRobotStateMsg(*robot_state_, ik_srv.request.ik_request.robot_state);
+  moveit::core::robotStateToRobotStateMsg(*robot_state_, ik_srv->ik_request.robot_state);
 
   // Load the poses into the request in difference places depending if there is more than one or not
-  geometry_msgs::PoseStamped ik_pose_st;
+  geometry_msgs::msg::PoseStamped ik_pose_st;
   ik_pose_st.header.frame_id = base_frame_;
   if (tip_frames_.size() > 1)
   {
     // Load into vector of poses
     for (std::size_t i = 0; i < tip_frames_.size(); ++i)
     {
       ik_pose_st.pose = ik_poses[i];
-      ik_srv.request.ik_request.pose_stamped_vector.push_back(ik_pose_st);
-      ik_srv.request.ik_request.ik_link_names.push_back(tip_frames_[i]);
+      ik_srv->ik_request.pose_stamped_vector.push_back(ik_pose_st);
+      ik_srv->ik_request.ik_link_names.push_back(tip_frames_[i]);
     }
   }
   else
   {
     ik_pose_st.pose = ik_poses[0];
 
     // Load into single pose value
-    ik_srv.request.ik_request.pose_stamped = ik_pose_st;
-    ik_srv.request.ik_request.ik_link_name = getTipFrames()[0];
+    ik_srv->ik_request.pose_stamped = ik_pose_st;
+    ik_srv->ik_request.ik_link_name = getTipFrames()[0];
   }
 
-  ROS_DEBUG_STREAM_NAMED("srv", "Calling service: " << ik_service_client_->getService());
-  if (ik_service_client_->call(ik_srv))
+  RCLCPP_DEBUG(LOGGER, "Calling service: %s", ik_service_client_->get_service_name());
+  auto result_future = ik_service_client_->async_send_request(ik_srv);
+  auto response = result_future.get();
+  if (rclcpp::spin_until_future_complete(node_, result_future) == rclcpp::executor::FutureReturnCode::SUCCESS)
   {
     // Check error code
-    error_code.val = ik_srv.response.error_code.val;
+    error_code.val = response->error_code.val;
     if (error_code.val != error_code.SUCCESS)
     {
-      ROS_DEBUG_STREAM_NAMED("srv", "An IK that satisifes the constraints and is collision free could not be found."
-                                        << "\nRequest was: \n"
-                                        << ik_srv.request.ik_request << "\nResponse was: \n"
-                                        << ik_srv.response.solution);
+      // TODO (JafarAbdi) Print the entire message for ROS2?
```

Comment:
yes, why not?

Reply:
The msgs in ROS2 don't have ostream operator overload so I have to pass through each [member](https://github.com/ros-planning/moveit_msgs/blob/master/msg/PositionIKRequest.msg), do you want me to add it anyway .?